assets/src/dashboard/app/index.js

```diff
@@ -58,21 +59,49 @@ import {
   TemplateDetailsView,
   ToasterView,
 } from './views';
+import useApi from './api/useApi';
 
 const AppContent = () => {
   const {
-    state: { currentPath },
+    state: {
+      currentPath,
+      queryParams: { id: templateId },
+    },
   } = useRouteHistory();
+  const { currentTemplate } = useApi(
+    ({
+      state: {
+        templates: { templates },
+      },
+    }) => ({
+      currentTemplate:
+        templateId !== undefined ? templates[templateId]?.title : undefined,
+    })
+  );
 
   useEffect(() => {
-    const dynamicPageTitle = ROUTE_TITLES[currentPath] || ROUTE_TITLES.DEFAULT;
-    window.document.title = sprintf(
+    let dynamicPageTitle = ROUTE_TITLES[currentPath] || ROUTE_TITLES.DEFAULT;
+    if (currentPath.includes(APP_ROUTES.TEMPLATE_DETAIL) && currentTemplate) {
+      dynamicPageTitle = sprintf(
+        /* translators: %s: Template name. */
+        __('Template: %s', 'web-stories'),
+        currentTemplate
+      );
+    }
+
+    document.title = sprintf(
       /* translators: Admin screen title. 1: Admin screen name, 2: Network or site name. */
       __('%1$s \u2039 %2$s \u2212 WordPress', 'web-stories'),
       dynamicPageTitle,
       ADMIN_TITLE
     );
-  }, [currentPath]);
+
+    if (currentPath.includes(APP_ROUTES.TEMPLATE_DETAIL) && currentTemplate) {
+      trackScreenView(dynamicPageTitle);
+    } else if (!currentTemplate) {
+      trackScreenView(dynamicPageTitle);
+    }
+  }, [currentPath, currentTemplate]);
```

Comment:
@BrittanyIRL You originally implemented the document title changes in #3394

Here I am trying to build upon that to have better titles when viewing a single template, and then tracking the updated title in telemetry. All while trying to avoid firing multiple tracking events if only either `currentPath` _or_ `currentTemplate` has updated.

Would love to get your feedback here whether this looks reasonable or if there's a better way. Thanks in advance! :)

Reply:
First glance this looks good to me, I have a few things I need to do this AM then I'll come back and play with this a little and get back to you. On my radar though!

Reply:
So I think this might be grabbing 1 extra tracking call because when I "close" a template it's removing the template id and then moving back to templates. So `Template Details` is getting through ` else if (!currentTemplate) {` when it's really just telling the view that since there's no id, let's go back to explore templates.

![page title update](https://user-images.githubusercontent.com/10720454/107814872-cd49fa00-6d2f-11eb-84ec-fc8ee24ea2c8.gif)

Reply:
could check that it's not `Template Details` in that `else if` to get around it or we could update how the template details are closed.

Reply:
Good catch, thanks for testing!

Revamped this a bit now and looks like I was able to fix it.